src/rpdk/core/project.py

```diff
@@ -217,15 +223,37 @@ def generate(self):
         # generate template for IAM role assumed by cloudformation
         # to provision resources if schema has handlers defined
         if "handlers" in self.schema:
+            handlers = self.schema["handlers"]
             template = self.env.get_template("resource-role.yml")
             permission = "Allow"
             path = self.root / ROLE_TEMPLATE_FILENAME
             LOG.debug("Writing Resource Role CloudFormation template: %s", path)
             actions = {
                 action
-                for handler in self.schema["handlers"].values()
+                for handler in handlers.values()
                 for action in handler.get("permissions", [])
             }
+
+            # calculate IAM role max session timeout based on highest handler timeout
+            # with some buffer (70 seconds per minute)
+            cud_handlers = [
+                handlers.get(handler, {})
+                for handler in ["create", "update", "delete"]
+                if handlers.get(handler, None)
```

Comment:
if we are going to only use the cud handlers, could we change to something like this? something about the current one seems a little roundabout. not a huge deal but seems cleaner
```
max(
(
handler.get("timeoutInMinutes", DEFAULT_ROLE_TIMEOUT_MINUTES)
for operation, handler in handlers.items() if operation in ["create", "update", "delete]
)
, 
default.....
)
```

Reply:
ah yeah, that is better 👍